src/utils/events.coffee

```diff
@@ -13,61 +13,40 @@
 
-        if data.length
-          columnWidth = this.getBestColumnWidth(axes, dimensions, data, options)
-          this.drawData(svg, dimensions, axes, data, columnWidth, options, handlers, dispatch)
-
-      setZoom: (svg, dimensions, axes, data, columnWidth, options, handlers, dispatch) ->
+      getZoomHandler: (svg, dimensions, axes, data, columnWidth, options, handlers, dispatch, zoom) ->
         self = this
-        
-        zoomHandler = () ->
-            zoomed = false
 
-            if options.axes.x.zoomable?
-              svg.selectAll('.x.axis').call(axes.xAxis)
-              zoomed = true
+        return ->
+          zoomed = false
 
-            if options.axes.y.zoomable?
-              svg.selectAll('.y.axis').call(axes.yAxis)
+          [ 'x', 'y', 'y2' ].forEach (axis) ->
+            if options.axes[axis]?.zoomable?
+              svg.selectAll(".#{axis}.axis").call(axes["#{axis}Axis"])
               zoomed = true
 
-            if options.axes.y2?.zoomable?
-              svg.selectAll('.y2.axis').call(axes.y2Axis)
-              zoomed = true
-
-            if data.length
-              columnWidth = self.getBestColumnWidth(axes, dimensions, data, options)
-              self.drawData(svg, dimensions, axes, data, columnWidth, options, handlers, dispatch)
-
-            if zoomed
-              self.createZoomResetIcon(svg, dimensions, axes, data, columnWidth, options, handlers, dispatch, zoom)
+          if data.length
+            columnWidth = self.getBestColumnWidth(axes, dimensions, data, options)
+            self.drawData(svg, dimensions, axes, data, columnWidth, options, handlers, dispatch)
 
-        zoom = this.getZoomListener(axes, options, zoomHandler)
-        svg.call(zoom)
+          if zoom and zoomed
+            self.createZoomResetIcon(svg, dimensions, axes, data, columnWidth, options, handlers, dispatch, zoom)
 
-      getZoomListener: (axes, options, zoomHandler) ->
-
-        zoomListener = d3.behavior.zoom()
+      setZoom: (svg, dimensions, axes, data, columnWidth, options, handlers, dispatch) ->
+        zoom = this.getZoomListener(axes, options)
 
-        if zoomHandler?
-          zoomListener.on("zoom", zoomHandler)
+        if zoom
+          zoom.on("zoom", this.getZoomHandler(svg, dimensions, axes, data, columnWidth, options, handlers, dispatch, zoom))
+          svg.call(zoom)
 
-        if options.axes.x?.zoomable?
-          zoomListener.x(axes.xScale)
+      getZoomListener: (axes, options) ->
+        zoomable = false
+        zoom = d3.behavior.zoom()
 
-        if options.axes.y?.zoomable?
-          zoomListener.y(axes.yScale)
-        
-        if options.axes.y2?.zoomable?
-          zoomListener.y(axes.y2Scale)
+        [ 'x', 'y', 'y2' ].forEach (axis) ->
+          if options.axes[axis]?.zoomable?
+            zoom[axis](axes["#{axis}Scale"])
+            zoomable = true
 
-        return zoomListener
\ No newline at end of file
+        return zoomable and zoom
```

Comment:
can we make this a little bit more expressive because it is hard to read, sth like `return zoom ? zoomable : false` or some coffeescript equivalent with `?`